### PR TITLE
a wild paper-submenu appears!

### PR DESCRIPTION
**Pull request by notwaldorf**
Look at it go!
screen shot 2015-08-07 at 3 59 55 pm

### CODE REVIEW

File: paper-submenu.html [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        notify: true,
+        observer: '_openedChanged'
+      }
+    },
+
+    behaviors: [
+      Polymer.IronControlState
+    ],
+
+    get _parent() {
```
Comment (cdata) [inline paper-submenu.html]: `protected` implementation should get docs ;)

File: test/paper-submenu.html [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          assert.isFalse(collapse3.opened);
+
+          MockInteractions.tap(trigger3);
+
+          assert.isFalse(collapse1.opened);
+          assert.isFalse(collapse2.opened);
+          assert.isFalse(collapse3.opened);
+        });
+
+        test('selecting and item styles it and the parent', function() {
```
Comment (cdata) [inline test/paper-submenu.html]: "selecting _an_ item"

File: paper-menu-shared.css [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  position: relative;
+  outline: 0;
+
+  @apply(--paper-menu-focused-item);
+}
+
+.selectable-content > ::content > *:focus:after {
+  @apply(--layout-fit);
+  background: currentColor;
+  /* FIXME move to paper-styles for next widget */
```
Comment (cdata) [inline paper-menu-shared.css]: Should this get fixed now?

File: paper-submenu.html [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+          <paper-item>Disabled 1</paper-item>
+          <paper-item>Disabled 2</paper-item>
+        </paper-menu>
+      </paper-submenu>
+    </paper-menu>
+
+Just like in `<paper-menu>`, the focused item is highlighted, and the selected
+item has bolded text. Please see the `<paper-menu>` docs for which attributes
+(such as `multi` and `selected`), and styling options are available for the menu.
+
```
Comment (cdata) [inline paper-submenu.html:50]: What about any relevant mixin / custom property docs?

File: paper-submenu.html [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    _openedChanged: function(opened, oldOpened) {
+      if (opened) {
+        this.fire('paper-submenu-open');
+      } else if (oldOpened != null) {
+        this.fire('paper-submenu-close');
+      }
+    },
+
+    _onParentIronActivate: function(event) {
+      if (Polymer.Gestures.findOriginalTarget(event) != this._parent) {
```
Comment (cdata) [inline paper-submenu.html]: This is a little bike sheddy, but I consider the semantics of strict equality checking to be superior in most cases (exception: checking for `!= null`).

**Comment by cdata** [comment]
One other thing, which may or may not make sense for this element: if `disabled` changes, you may want to auto-collapse the submenu (if it becomes disabled). What do you think is the right thing to do?

**Comment by cdata** [comment]
(We do this in `paper-menu-button`, for example: https://github.com/PolymerElements/paper-menu-button/blob/master/paper-menu-button.html#L330-L341)

**Comment by notwaldorf** [comment]
@cdata done all the things!

File: paper-submenu.html [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+    behaviors: [
+      Polymer.IronControlState
+    ],
+
+    get __parent() {
+      return Polymer.dom(this).parentNode;
+    },
+
+    get __trigger() {
```
Comment (cdata) [inline paper-submenu.html:113]: Wow. Such private. Very underscore.

**Comment by cdata** [comment]
LGTM :+1:

**Pull request merged**